Tribler/community/tunnel/hidden_community.py

```diff
@@ -312,26 +397,34 @@ def on_key_request(self, messages):
 
     def check_key_response(self, messages):
         for message in messages:
-            if not message.source.startswith(u"circuit_"):
```

Comment:
I'm going to try, lets imagine Alice is going to request the key of Bob, which has an intro-point at Charlie. Alice is going to use her pre-existing circuit she made with Dean.

```
A ---> D ---> C ---> B (key-request)
A <--- D <--- C <--- B (key-response)
```

But there was no code, which would cause Dean to forward the key-response message back to Alice right? Let me explain how it should have worked.

Charlie is the receiver of the key-request message. He receives this message over its dispersy port, as that's the port he used to announce to the DHT to. But he's receiving this message from the exit-socket of Dean (message.candidate.sock_addr). This sock_addr is put into the message as the source, and the key-request is forwarded to Bob. 

Bob can now reply to Alice, by simply instructing Charlie to exit-data to the sock_addr he received as the source. This is still the ExitSocket that Dean constructed for the circuit he has with Alice. And hence, any packet which is received over this socket, is automatically forwarded to her.